content/docs/components-and-props.md

```diff
@@ -78,28 +77,28 @@ ReactDOM.render(
 
 [](codepen://components-and-props/rendering-a-component)
 
-Let's recap what happens in this example:
+चलिए देकते है इस उदाहरण में क्या होता है:
 
-1. We call `ReactDOM.render()` with the `<Welcome name="Sara" />` element.
-2. React calls the `Welcome` component with `{name: 'Sara'}` as the props.
-3. Our `Welcome` component returns a `<h1>Hello, Sara</h1>` element as the result.
-4. React DOM efficiently updates the DOM to match `<h1>Hello, Sara</h1>`.
+1. हम `<Welcome name="Sara" />` के द्वारा `ReactDOM.render()` को call karte है|
+2. React `Welcome` कौम्पोनॅन्ट को `name: 'Sara'}` के साथ props के रूप मे call करती है|
+3. हमारा `Welcome` कौम्पोनॅन्ट परिणाम के तौर पर वापस करता है `<h1>नमस्ते, Sara</h1>` एलिमेंट|
+4. React DOM कुशलता पूर्वक DOM को `<h1>नमस्ते, Sara</h1>` से अपडेट करता है|
 
->**Note:** Always start component names with a capital letter.
+>**ध्यान दें:** हमेशा रेंकौम्पोनॅन्टस की नाम की शुरुवात कैपिटल लेटर से होनी चाहिए|
 >
->React treats components starting with lowercase letters as DOM tags. For example, `<div />` represents an HTML div tag, but `<Welcome />` represents a component and requires `Welcome` to be in scope.
+>React उन कौम्पोनॅन्टस को DOM टैग्स मानता है जो लोअरकेस से स्टार्ट होते है| उदाहरण के लिए, `<div />` HTML div टैग्स को दर्शाता है, पर `<Welcome />` एक कौम्पोनॅन्ट को दर्शाता है और इसको Welcome स्कोप में चाहिए होता है|
 >
->To learn more about the reasoning behind this convention, please read [JSX In Depth](/docs/jsx-in-depth.html#user-defined-components-must-be-capitalized).
+>इसके पीछे की सोच के बारे में और गहराई में जानने के लिए, कृपया [JSX गहराई में](/docs/jsx-in-depth.html#user-defined-components-must-be-capitalized) पढ़ें|
 
-## Composing Components {#composing-components}
+## कौम्पोनॅन्टस की रचना {#composing-components}
 
-Components can refer to other components in their output. This lets us use the same component abstraction for any level of detail. A button, a form, a dialog, a screen: in React apps, all those are commonly expressed as components.
+कौम्पोनॅन्टस अपने आउटपुट में अन्य कौम्पोनॅन्टस का उल्लेख कर सकते हैं| इससे हम विस्तार के किसी भी स्तर के लिए उन्ही कौम्पोनॅन्ट का उपयोग कर सकते हैं| एक बटन, एक फॉर्म, एक डायलॉग, एक स्क्रीन React ऍप्स में, इन सबको आमतौर पर कौम्पोनॅन्टस के रूप में व्यक्त किया जाता है|
```

Comment:
इससे हम विस्तार के किसी भी स्तर के लिए उन्ही कौम्पोनॅन्ट का उपयोग कर सकते हैं -> unclear sentence?

Reply:
एक स्क्रीन: in react apps me sabko